src/imitation/rewards/reward_nets.py

```diff
@@ -318,6 +302,210 @@ def forward(self, state, action, next_state, done):
         return outputs
 
 
+class NormalizedRewardNet(RewardNetWrapper):
+    """A reward net that normalizes the output of its base net.
+
+    Only requires the implementation of a forward pass (calculating rewards given
+    a batch of states, actions, next states and dones) as it inherits RewardNet.
+    """
+
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        base: RewardNet,
+        rew_normalize_class: Type[nn.Module],
+        normalize_images: bool = True,
+    ):
+        """Initialize the NormalizedRewardNet.
+
+        Args:
+            observation_space: the observation space of the environment
+            action_space: the action space of the environment
+            base: a base RewardNet
+            rew_normalize_class: The class to use to normalize rewards. This
+                can be any nn.Module that preserves the shape; e.g. `nn.Identity`,
+                `nn.BatchNorm*`, `nn.LayerNorm`, or `networks.RunningNorm`.
+            normalize_images: passed through to `RewardNet.__init__`,
+                see its documentation
+        """
+        super().__init__(
+            observation_space=observation_space,
+            action_space=action_space,
+            base=base,
+            normalize_images=normalize_images,
+        )
+        # assuming reward is always a scalar
+        self.rew_normalize_layer = (
+            rew_normalize_class(1) if rew_normalize_class else None
```

Comment:
I think `rew_normalize_class` should always be supplied (based on type annotation, and why wrap with a normalization layer if you don't want to normalize?) so can simplify this expression

Reply:
Makes sense.